enhancements/network/infiniband.md

```diff
@@ -0,0 +1,123 @@
+---
+title: infiniband
+authors:
+  - "@zshi"
+reviewers:
+  - "@pliurh"
+  - "@dougbtv"
+  - "@s1061123"
+  - "@danwinship"
+  - "@squeed"
+  - "@dcbw"
+  - "@yrobla"
+approvers:
+  - "@knobunc"
+  - "@fepan"
+creation-date: 2020-05-06
+last-updated: 2020-05-06
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Infiniband
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+RDMA (Remote Direct Memory Access) fabrics such as Infiniband and RDMA over
+Converged Ethernet (RoCE) are key to unlocking scalable performance for
+demanding Artificial Intelligence (AI) and Machine Learning (ML) applications.
+RoCE v2 is supported in SR-IOV Operator as Tech Preview since 4.3, This
+enhancement is to introduce native Infiniband support.
+
+## Motivation
+
+Enable high-performance message transmission via RDMA for distributed workloads
+such as AI/ML.
+
+### Goals
+
+- Provide native Infiniband support on pod additional network interfaces
+
+### Non-Goals
+
+- Support RoCE v2
+- Support GPUDirect RDMA
+- Support Ethernet over Infiniband such as IPoIB (IP over Infiniband)
+- Support Infiniband on pod primary network interface
+
+## Proposal
+
+Extend SR-IOV Operator to support both Ethernet and Infiniband devices.
+
+SR-IOV Hardware NIC is managed by SR-IOV Operator and will be provisioned
+with a new device type `infiniband` or `ib`.
+
+A new Container Network Interface (CNI) plugin `ib-sriov-cni` will be added.
```

Comment:
if we support only this then the use will need to manually add InfiniBand GUID to pod spec. the ib-kubernetes allow to allocate InfiniBand  GUID similar to what kubemacpool. 

The ib-kubernetes can also configure the pkeys (which are similar to vlan) but this will required UFM. We can disable it by using noop plugin so we can just use the InfiniBand  GUD allocation.

Reply:
@moshe010 Is that possible to have a default random GUID automatically without other tools? Or we have to inject the GUID to the pod annotations in any way?

Reply:
Without ib-kubernetes allocating GUID, I think user can specify it in pod annotation manually.
The question I have is how useful it would be to support infiniband without Pkeys?

Reply:
Yes use will need to inject it by himself or we can we can  default random GUID automatically (but this may cause collisions in the cluster).

Pkeys are useful if you want isolation (similar concept to vlan in ethernet) so it depend on the customer use-case.

Reply:
we can do default random GUID automatically  in the SR-IOV operator but may have collisions in the cluster. 
with the current design use can inject it in the pod annotation
so pkeys allow isolation (same as vlan in ethernet) so I think it depend on use-case. How many customers use vlan in SR-IOV cni? is it common?

Reply:
I don't have concrete info on whether pkey partiion is used, will need to ask.
but vlan in SR-IOV cni is commonly used.

Reply:
There are several considerations in adding ib-kubernetes support:
- UFM is not supported on RHEL
- Part of UFM functions are included in opensm which needs to be included in RHCOS or delivered via container image.
- Will ib-kubernetes be able to add opensm as a plugin?
- Needs to find equivalent tools on RHEL to the rest of UFM functions such as monitoring, statistics.

Reply:
We can add opesm plugin but I am not sure what inbox opensm support. We can provide opensm that comes with ofed which we know should work.

Reply:
the upstream opensm doesn't support SR-IOV only the opensm which comes with Mellanox OFED